spec/features/users/logout_spec.rb

```diff
@@ -0,0 +1,108 @@
+      @paper = @mike.items.create(name: "Lined Paper", description: "Great for writing on!", price: 20, image: "https://cdn.vertex42.com/WordTemplates/images/printable-lined-paper-wide-ruled.png", inventory: 25)
+      @pencil = @mike.items.create(name: "Yellow Pencil", description: "You can write on paper with it!", price: 2, image: "https://images-na.ssl-images-amazon.com/images/I/31BlVr01izL._SX425_.jpg", inventory: 100)
+
+      visit "/items/#{@paper.id}"
+
+      click_on "Add To Cart"
+
+      within(".topnav") do
+        expect(page).to have_link("Login")
+        click_link "Login"
+      end
+    end
+
+    it "Can logout a regular logged-in user" do
+      fill_in :email, with: @user.email
+      fill_in  :password, with: @user.password
+
+      click_button "Login"
+
+      within(".topnav") do
+        expect(page).to have_link("Cart: 1")
+      end
+
+      within(".topnav") do
+        expect(page).to have_link("Logout")
+        expect(page).to_not have_link("Login")
+        expect(page).to_not have_link("Register")
+        click_link "Logout"
+      end
+
+      expect(current_path).to eq("/")
+      expect(page).to have_content("Congrats, you're leaving, but you should stay!")
+      within(".topnav") do
+        expect(page).to have_link("Cart: 0")
+        click_link "Login"
+      end
+
+      expect(current_path).to eq('/login')
+    end
+
+    it "Can logout a merchant logged-in user" do
+      fill_in :email, with: @merchant.email
+      fill_in  :password, with: @merchant.password
+
+      click_button "Login"
+
+      within(".topnav") do
+        expect(page).to have_link("Cart: 1")
+      end
+
+      within(".topnav") do
+        expect(page).to have_link("Logout")
+        expect(page).to_not have_link("Login")
+        expect(page).to_not have_link("Register")
+        click_link "Logout"
+      end
+
+      expect(current_path).to eq("/")
+      expect(page).to have_content("Congrats, you're leaving, but you should stay!")
+      within(".topnav") do
+        expect(page).to have_link("Cart: 0")
+        click_link "Login"
+      end
+
+      expect(current_path).to eq('/login')
+    end
+
+    it "Can logout a admin logged-in user" do
+      fill_in :email, with: @admin.email
+      fill_in  :password, with: @admin.password
+
+      click_button "Login"
+
+      within(".topnav") do
+        expect(page).to have_link("Cart: 1")
+      end
+
+      within(".topnav") do
+        expect(page).to have_link("Logout")
+        expect(page).to_not have_link("Login")
+        expect(page).to_not have_link("Register")
+        click_link "Logout"
+      end
+
+      expect(current_path).to eq("/")
+      expect(page).to have_content("Congrats, you're leaving, but you should stay!")
+      within(".topnav") do
+        expect(page).to have_link("Cart: 0")
+        click_link "Login"
+      end
+
+      expect(current_path).to eq('/login')
+    end
+  end
```

Comment:
Might be some way to dry this up some since there's a lot of duplication, but maybe not.  For a future day for sure.

Reply:
Yes, cheers and good suggestion.